core/src/main/scala/filodb.core/metadata/Dataset.scala

```diff
@@ -26,31 +25,30 @@ import filodb.memory.format.{BinaryVector, RowReader, TypedIterator, ZeroCopyUTF
  *   partition columns:  metricName:string, tags:map
  *   data columns:       timestamp:long, value:double
  *
+ * NOTE: this data structure will be deprecated slowly in favor of PartitionSchema/DataSchema.
+ * NOTE2: name is used for ingestion stream name, which is separate from the name of the schema.
+ *
  * The Column IDs (used for querying) for data columns are numbered starting with 0, and for partition
  * columns are numbered starting with PartColStartIndex.  This means position is the same or easily derived
  *
  * The rowKeyIDs are the dataColumns IDs/positions for the "row key", typically a timestamp column but
  * something which makes a value unique within a partition and describes a range of data in a chunk.
  */
-final case class Dataset(name: String,
-                         partitionColumns: Seq[Column],
-                         dataColumns: Seq[Column],
-                         rowKeyIDs: Seq[Int],
-                         downsamplers: Seq[ChunkDownsampler],
-                         database: Option[String] = None,
-                         options: DatasetOptions = DatasetOptions.DefaultOptions) {
-  require(rowKeyIDs.nonEmpty)
-  val ref = DatasetRef(name, database)
-  val rowKeyColumns   = rowKeyIDs.map(dataColumns)
-  val rowKeyRouting   = rowKeyIDs.toArray
-
-  val ingestionSchema = RecordSchema2.ingestion(this)  // TODO: add predefined keys yo
-  val comparator      = new RecordComparator(ingestionSchema)
-  val partKeySchema   = comparator.partitionKeySchema
+final case class Dataset(name: String, schema: Schema) {
+  def options: DatasetOptions = schema.partition.options
+  def dataColumns: Seq[Column] = schema.data.columns
+  def partitionColumns: Seq[Column] = schema.partition.columns
```

Comment:
Why def for the three above?

Reply:
🤷🏻‍♂️  I could change it to val.  def uses no memory, but that doesn't matter here.  They are quick references, maybe val is slightly faster?